### PR TITLE
Improve focus styling

### LINKED ISSUE #15423: Some focus indicators have poor visibility in dark mode
## Description

Some (default?) focus indicators like these three are poorly visible and inconsistent with other focus indicators we use

Screenshot from 2023-11-17 18-23-57
Screenshot from 2023-11-17 18-27-58
Screenshot from 2023-11-17 18-23-42

## Expected behavior

Focus indicators use the same, recognisable and contrastive style (colour/borders width/round radius) across application. Like:

image

image

## Context

- Operating System and version: Ubuntu
- Browser and version: Chrome
- JupyterLab version: 4.1.0a03

### PR DESCRIPTION
**Pull request by fcollonval**
## References

Fixes following analysis of https://github.com/jupyterlab/jupyterlab/pull/16438

Fixes #15423 

## Code changes

More homogeneous focus highlight with a 2px border/outline and common color.
`--jp-accept-color-active` has a very poor contrast with black in dark high contrast theme 2.43. The new one has contrast of 6.96.

Bump the toolkit to 0.16.3 - it adds some margin to make space around the components for the focus highlight

Change margin/padding on sidebar tabs, accordion titles and toolbar items to allow the full display of the focus highlight

Improve caret position for select element.

Improve focus for dock panel tab and add button

> The add button bottom is not highlighted because the border takes the full remaining width of the tab bar, not only the bottom of the button.

## User-facing changes

| Before | After |
| --- | --- |
|current_focus_1| after_focus_1|
| before_focus_2| after_focus_2|
|Screenshot from 2023-11-17 18-23-57|image |
| | image |
| | image |

> Snapshot taken in Chromium 125

## Backwards-incompatible changes

None

### CODE REVIEW

**Comment by fcollonval** [comment]
bot please update snapshots

**Comment by krassowski** [comment]
Worth checking how it now looks in the dark mode: https://github.com/jupyterlab/jupyterlab/issues/15423

**Comment by fcollonval** [comment]
bot please update snapshots

**Comment by fcollonval** [comment]
bot please update documentation snapshots

File: packages/ui-components/src/components/search.tsx [hunk 1]
```diff
@@ -172,23 +172,26 @@ export const FilterBox = (props: IFilterBoxProps): JSX.Element => {
...
+      );
+    },
+    [props.updateFilter, props.useFuzzyFilter, props.caseSensitive]
+  );
 
   return (
     <Search
       className="jp-FilterBox"
-      ref={props.inputRef}
+      ref={props.inputRef as any}
```
Comment (krassowski) [inline packages/ui-components/src/components/search.tsx:194]: Can we replace `any` with the correct typing here?

**Review by krassowski — COMMENTED** [review]
Looks good, just one question

File: packages/ui-components/src/components/search.tsx (hunk 1 above)
Comment (fcollonval) [reply packages/ui-components/src/components/search.tsx:194]: The webcomponent does not exactly matches a HTMLInputElement. But I can improve it a bit.

**Comment by krassowski** [comment]
This already has some conflicts, and once we merge https://github.com/jupyterlab/jupyterlab/pull/16446 there will be more conflicts. Maybe we could first get https://github.com/jupyterlab/jupyterlab/pull/16446 over the finish line and then get this one in?

**Comment by fcollonval** [comment]
bot please update snapshots

File: galata/test/documentation/custom_css.test.ts-snapshots/custom-css-main-documentation-linux.png
Comment (krassowski) [inline galata/test/documentation/custom_css.test.ts-snapshots/custom-css-main-documentation-linux.png:1]: This looks like a flaky update? Probably fine as it is passing with it too

Comment (fcollonval) [reply galata/test/documentation/custom_css.test.ts-snapshots/custom-css-main-documentation-linux.png:1]: It looks like it. But if it is ok for you, I rather merge as is.

File: packages/running/style/base.css [hunk 2]
```diff
@@ -181,29 +177,6 @@ span.jp-RunningSessions-icon > svg {
...
-  box-shadow: none;
-  background-color: var(--jp-layout-color2);
-}
-
-.jp-RunningSessions-shutdownAll.jp-ToolbarButtonComponent[disabled] {
-  color: var(--jp-ui-font-color2);
-}
-
-.jp-RunningSessions-shutdownAll.jp-ToolbarButtonComponent[disabled]:hover {
-  background: none;
```
Comment (krassowski) [inline packages/running/style/base.css:206]: Is this intentional? I think that previous version with grey color and without hover background for disabled button was better

image

**Review by krassowski — COMMENTED** [review]
Tested locally, looks great to me except for the `.jp-RunningSessions-shutdownAll.jp-ToolbarButtonComponent[disabled]` styles which I think would be better kept.

**Comment by fcollonval** [comment]
bot please update documentation snapshots

**Comment by fcollonval** [comment]
bot please update galata snapshots

**Comment by fcollonval** [comment]
Merging to move forward on the treeview PR #16488

**Pull request merged**